### ISSUE TITLE
Display friendly error on too big avatar image

### ISSUE THREAD

**Issue by Zaczero** [Zaczero/openstreetmap-ng#31]
The goal is to:
- Improve handling of too big files in the frontend
- Catch decompression errors on backend

**Comment by SaadQasim7** [comment]
is it still active

**Comment by Zaczero** [comment]
Yup. The library that performs the decompression is opencv (imdecode method). The error messages can be added to the extra_en.yaml translations file. Basically we want a human readable error message when someone uploads a too big file in megapixels.

**Comment by KonradowyPL** [comment]
yes!

**Comment by hobyt-aluzar** [comment]
I can take a focused pass at this: map Pillow decompression-bomb failures to the existing image-too-big response, map unreadable image data to a friendly 422 response, add lightweight client-side file-size validation for avatar/background uploads, and cover the backend cases with focused tests.

**Comment by MapleQiAN** [comment]
I am starting a focused fix for this: catch image decompression failures and surface a friendly validation error for oversized/unreadable uploads, then add targeted backend coverage for the error mapping. I will keep the change narrowly scoped to issue #31.

**Comment by MapleQiAN** [comment]
Opened PR #250 with the backend fix and targeted coverage: https://github.com/openstreetmap-ng/openstreetmap-ng/pull/250

**Comment by loong-solvable** [comment]
Opened PR #253 for the frontend side of this issue: selected avatar, app avatar, and background images now get a standard form error before submit when their dimensions exceed the configured limits. This complements the backend decode/decompression handling in #250.

**Comment by Adysekus** [comment]
PR #281

**Comment by esghound3-maker** [comment]
Hi, is the $10 bounty for this issue still available? I’d like to work on improving the too-large avatar upload error
 handling.

**Comment by danny101488** [comment]
I have a local implementation patch ready for this issue. The normal fork/PR path is still blocked on my side, so I published the patch as a temporary review artifact here:

https://github.com/danny101488/Bubio-testing/blob/main/codex-earnings/osmng/osmng-31-avatar-error.patch

Summary:
- catches Pillow decode/exif failures, decompression-bomb failures, corrupt animation extraction, and save/optimization errors
- maps those failures to the existing image-too-large API error path instead of surfacing a raw server error
- clears the avatar file input after upload errors so users can reselect the same/fixed file
- includes a regression test for invalid avatar bytes returning the expected API error status

Local verification: `git diff --check` passed. I did not run the full suite because my local machine does not have the repo's Python 3.14/Nix test environment installed, and I avoided dependency-fetching setup on this machine.

I will open a proper PR once I have a usable fork/branch path; sharing the patch now so it is reviewable meanwhile.

**Comment by exal-gh-33** [comment]
Submitted PR #375 for this bounty: https://github.com/openstreetmap-ng/openstreetmap-ng/pull/375

**Comment by AxisIV** [comment]
Submitted PR #374 for this bounty: https://github.com/openstreetmap-ng/openstreetmap-ng/pull/374. The latest run is green across Ubuntu, cython, macOS, and Socket checks.
